### ISSUE TITLE
Is there an example of a react_on_rails Docker integration?

### ISSUE THREAD

**Issue by vassyz** [shakacode/react_on_rails#691]
I can't get the app to work with my limited knowledge of Docker. Is there a working example in the wild? I might remember it wrong, but I think at some point there was a pull request that tried adding a docs entry regarding Docker.

**Comment by justin808** [comment]
@vassyz https://github.com/shakacode/react-webpack-rails-tutorial/pull/332

maybe you can finish this up?

**Comment by bloomdido** [comment]
I think this really needs to be split up into documentation/gists/sample projects because everyone's Docker/deployment configuration is going to be different. I have spent the last week getting my Docker configuration working and discovered more than a few different paths.

From https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/

> In almost all cases, you should only run a single process in a single container. Decoupling applications into multiple containers makes it much easier to scale horizontally and reuse containers. If that service depends on another service, make use of container linking.

In practice, this typically means building a single (app) container running Puma or some other app engine. While it is certainly possible to deploy an app into production as just a single Puma instance (depending on your Ruby implementation and the use of concurrency), a more typical setup is a number of Puma instances (app containers) behind some sort of proxy & load-balancer like nginx (web container). In that situation, the web container would more likely be serving your assets, meaning you would either need to expose them as a volume from your app container, or copy them into your web container when it is built. Personally, I chose the latter. That means my assets must be compiled BEFORE either my app or web containers are built. In this case, there is no need to include webpack (or foreman) inside the app container.

Then there are other server configurations, such as Passenger. And all of this differs for a development environment, where you would want your code mounted as a volume in your container (so it would not have to be rebuilt after every change). And where and how your eventually intend to deploy (Docker Swarm, AWS EC2 Container Service, Elastic Beanstalk, etc.)

I know exactly how complex this topic can be, because I've spent the last week tracking down every little bit of information I could regarding best practices regarding deploying just a ordinary Rails app, let alone one using ReactOnRails. The information is scarce and often outdated (Docker is changing constantly).

Anyway, I am pretty busy getting my project up and can't promise I will have much time in the coming weeks/months to contribute, but I will try to do my best.
